src/sentry/integrations/vercel/webhook.py

```diff
@@ -44,4 +54,153 @@ def post(self, request):
+            return self.respond({"detail": "Integration not found"}, status=404)
+
+        # for each org integration, search the configs to find one that matches the vercel project of the webhook
+        for org_integration in org_integrations:
+            project_mappings = org_integration.config.get("project_mappings") or []
+            matched_mappings = filter(lambda x: x[1] == vercel_project_id, project_mappings)
+            if matched_mappings:
+                organization = org_integration.organization
+                sentry_project_id = matched_mappings[0][0]
+
+                logging_params["sentry_project_id"] = sentry_project_id
+
+                # look up the project so we can get the slug
+                try:
+                    project = Project.objects.get(id=sentry_project_id)
+                except Project.DoesNotExist:
+                    logger.info("Project not found", extra=logging_params)
+                    return self.respond({"detail": "Project not found"}, status=404)
+
+                logging_params["organization_id"] = organization.id
+
+                # find the connected sentry app installation
+                try:
+                    installation_for_provider = SentryAppInstallationForProvider.objects.select_related(
+                        "sentry_app_installation"
+                    ).get(
+                        organization_id=organization.id, provider=self.provider
+                    )
+                    sentry_app_installation = installation_for_provider.sentry_app_installation
+                except SentryAppInstallationForProvider.DoesNotExist:
+                    logger.info("Installation not found", extra=logging_params)
+                    return self.respond({"detail": "Installation not found"}, status=404)
+
+                logging_params["sentry_app_installation_id"] = sentry_app_installation.id
+
+                # find a token associated with the installation so we can use it for authentication
+                try:
+                    sentry_app_installation_token = (
+                        SentryAppInstallationToken.objects.select_related("api_token")
+                        .filter(sentry_app_installation=sentry_app_installation)
+                        .first()
+                    )
+                except SentryAppInstallationToken.DoesNotExist:
+                    logger.info("Token not found", extra=logging_params)
+                    return self.respond({"detail": "Token not found"}, status=404)
+
+                # find the commmit sha so we can  use it as as the release
+                commit_sha = (
+                    meta.get("githubCommitSha")
+                    or meta.get("gitlabCommitSha")
+                    or meta.get("bitbucketCommitSha")
+                )
+
+                # contruct the repo depeding what provider we use
+                if meta.get("githubCommitSha"):
+                    # There is also githubRepo and githubRepo
```

Comment:
> There is also githubRepo and githubRepo

not sure what this is supposed to be but seems like a typo, assuming the second `githbRepo` is something else

Reply:
@MeredithAnya good catch!